### DISCUSSION TITLE
Fear & Greed Index when running from source

### DISCUSSION THREAD

**Discussion by jokorn** [ghostfolio/ghostfolio#833] [Q&A] [answered]
Thank you for this great, ever-developing piece of software :)

One question: How do I make the Fear & Greed index available when running from source? On the live demo there is an icon at the bottom of the page that directs to https://ghostfol.io/home/market but this icon is not showing on my setup. I have cloned the repo and run Ghostfolio successfully using the instructions listed under "Development" in the readme file. It works great except for the missing Fear & Greed index - no errors or anything...

When I navigate to /home/market on my setup, this is all I see (again, there is no icon so I have to type the URL manually to get to this screen):
![image](https://user-images.githubusercontent.com/7759201/163601162-e2ae101a-89a3-4e37-8c0b-5955eaf2aa3d.png)

Hope that you can help and thanks again...

**Comment by dtslvr** [comment, marked as answer]
Hi @jokorn

The _Fear & Greed index_ is an unofficial feature with no documentation / support, because the setup is a bit tricky.

You need the following configuration to make it work on your local machine:

- An api key to access this endpoint: https://rapidapi.com/rpi4gx/api/fear-and-greed-index/
- `API_KEY_RAPID_API=<api-key-from-above>` in your `.env` file
- `ENABLE_FEATURE_FEAR_AND_GREED_INDEX=true` in your `.env` file
- The following entry in the database table `SymbolProfile`:
   ```
   {
     currency: "",
     dataSource: "RAPID_API",
     symbol: "_GF_FEAR_AND_GREED_INDEX"
   }
   ```

**Reply by jokorn** [reply]
Thank you for pointing me in the right direction. It almost works now except that the rapidapi endpoint seems to only return data from today, yesterday, 1 week ago, 1 month ago and 1 year ago:

![image](https://user-images.githubusercontent.com/7759201/163691640-897e0ba8-ee23-45eb-9806-7723f4a34411.png)
![image](https://user-images.githubusercontent.com/7759201/163691677-f36c6f3a-fea6-422b-967e-d711413b5f55.png)

I am not sure if this is the reason that the plot is not being drawn on my local machine:

![image](https://user-images.githubusercontent.com/7759201/163691704-ce8183df-2ea3-471a-b300-0c9ee5fef177.png)

(Thank you for changing this from "issue" to "discussion" - I completely missed the discussion section...)

**Reply by dtslvr** [reply]
That looks fine so far 😎 

It's a limitation of the API itself, the history is not fully available. You could get the missing data from the live demo (it's getting populated there for quite some time) and import it in your database. Either with a script or by clicking on the red dates and add it manually.

The chart should be rendered though, do you get an error in the console?

**Reply by jokorn** [reply]
There is this error in the console:
![image](https://user-images.githubusercontent.com/7759201/163725108-8e188714-1ca3-44b2-97b5-9b4a7b067c56.png)

Regarding the missing historic data, that is not important. I just wondered if it was the reason for the missing chart...

**Reply by dtslvr** [reply]
Thanks for reporting log @jokorn. I have created a little fix here #840.

**Reply by jokorn** [reply]
Works great after #840. Thanks a lot!

![image](https://user-images.githubusercontent.com/7759201/163849318-56a598be-f967-43c7-b562-813a29b53d8f.png)

**Reply by dtslvr** [reply]
Hi @jokorn

The configuration has changed as of Ghostfolio `1.204.1`: https://github.com/ghostfolio/ghostfolio/discussions/833#discussioncomment-2578057 (I have already updated my answer).

* `RAKUTEN_RAPID_API_KEY ` -> `API_KEY_RAPID_API` (`.env` file)
* `dataSource: "RAKUTEN"` -> `dataSource: "RAPID_API"` (`SymbolProfile` table)

**Comment by pvieira84** [comment]
Hello @dtslvr, has this changed again? I see that in this pull request https://github.com/ghostfolio/ghostfolio/pull/7103 there is a mention to this but I'm unable to understand what was done.

Since I've upgraded to v3.13.0, the chart is not loading anymore nor there is any market data being gathered from RapidAPI.
<img width="1362" height="977" alt="image" src="https://github.com/user-attachments/assets/be834cb7-fc7d-4a37-8763-f84e9a0a612c" />

**Reply by dtslvr** [reply]
The symbol recently changed from `_GF_FEAR_AND_GREED_INDEX` to `GF_FEAR_AND_GREED_INDEX_STOCKS`.

The feature no longer works because the [endpoint](https://rapidapi.com/rpi4gx/api/fear-and-greed-index) has been removed from _RapidAPI_.

**Reply by pvieira84** [reply]
@dtslvr there is another endpoint at RapidAPI which allows up to 20 free requests per month: https://rapidapi.com/apimaker/api/fear-and-greed-index2

Found also a completely free alternative: https://feargreedchart.com/api-docs

On another note, I have changed the symbol accordingly in the DB however the chart is still not rendering. Any idea why?

**Reply by dtslvr** [reply]
It is now possible to use any URL via the `MANUAL` data source (see the `DATA_SOURCE_FEAR_AND_GREED_INDEX_STOCKS` environment variable).

Regarding the chart: have you also adapted the symbol for the entries in the _MarketData_ table?

**Reply by pvieira84** [reply]
@dtslvr I also adjusted the symbol in the MarketData table
